crates/core_arch/src/x86/cpuid.rs

```diff
@@ -55,26 +55,33 @@ pub unsafe fn __cpuid_count(leaf: u32, sub_leaf: u32) -> CpuidResult {
     let ebx;
     let ecx;
     let edx;
+
     #[cfg(target_arch = "x86")]
     {
-        llvm_asm!("cpuid"
-                  : "={eax}"(eax), "={ebx}"(ebx), "={ecx}"(ecx), "={edx}"(edx)
-                  : "{eax}"(leaf), "{ecx}"(sub_leaf)
-                  : :);
+        asm!(
+            "cpuid",
+            inlateout("eax") leaf => eax,
+            lateout("ebx") ebx,
+            inlateout("ecx") sub_leaf => ecx,
+            lateout("edx") edx,
+            options(nostack, nomem, preserves_flags),
```

Comment:
Is an instruction that (among other things) acts as a fence actually `nomem`? IDK. I added it under the logic of: It shouldn't have a \"memory\" clobber.

But I suspect that's not quite what it means.

Reply:
You need to remove `nomem` for the `asm!` to act as a compiler fence. This will prevent memory accesses from being reordered across the `asm!`.

Reply:
Yeah, thanks.